### PR TITLE
refactor(frontend): re-adjust types for TransactionType to have a generic one

### PR DESCRIPTION
**Pull request by AntonioVentilii**
# Motivation

It makes sense that we have a generic `TransactionType` type, since we are going to have a component that can be used for all networks. And then, filter it for the other network-specific types.

# Changes

- Expand current `TransactionType` to have all types.
- Create `BtcTransactionType` type, filtering.
- Adjust `EthTransactionType` and `IcTransactionType` to filter too.
- Use zod for `TransactionType`.

### CODE REVIEW

File: src/frontend/src/btc/types/btc-transaction.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import type { TransactionType } from '$lib/types/transaction';
+
+export type BtcTransactionType = Exclude<
```
Comment (peterpeterparker) [inline src/frontend/src/btc/types/btc-transaction.ts:3]: Wouldn't `Pick` be more future prone here?

File: src/frontend/src/eth/types/eth-transaction.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import type { Transaction, TransactionType } from '$lib/types/transaction';
 
-export type EthTransactionType = TransactionType | 'withdraw' | 'deposit';
+export type EthTransactionType = Exclude<TransactionType, 'approve' | 'burn' | 'mint'>;
```
Comment (peterpeterparker) [inline src/frontend/src/eth/types/eth-transaction.ts:3]: ditto

File: src/frontend/src/icp/types/ic-transaction.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ export type IcrcTransaction = {
 
 export type IcTransaction = IcpTransaction | IcrcTransaction;
 
-export type IcTransactionType = TransactionType | 'approve' | 'burn' | 'mint';
+export type IcTransactionType = Exclude<TransactionType, 'withdraw' | 'deposit'>;
```
Comment (peterpeterparker) [inline src/frontend/src/icp/types/ic-transaction.ts:22]: and ditto ;)

**Comment by AntonioVentilii** [comment]
[OISY-130]

File: src/frontend/src/btc/types/btc-transaction.ts (hunk 1 above)
Comment (AntonioVentilii) [reply src/frontend/src/btc/types/btc-transaction.ts:3]: Yes, it is less confusing for sure! I was thinking that it was not really top code while coding eheh!

Didn't thought of `Pick`

File: src/frontend/src/eth/types/eth-transaction.ts (hunk 2 above)
Comment (AntonioVentilii) [reply src/frontend/src/eth/types/eth-transaction.ts:3]: same

File: src/frontend/src/icp/types/ic-transaction.ts (hunk 3 above)
Comment (AntonioVentilii) [reply src/frontend/src/icp/types/ic-transaction.ts:22]: same

File: src/frontend/src/lib/schema/transaction.schema.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+const icpSpecificTypes = ['approve', 'burn', 'mint'] as const;
+
+const allTypes = [...commonTypes, ...ethSpecificTypes, ...icpSpecificTypes] as const;
+
+export const btcTransactionTypes = commonTypes;
+
+export const ethTransactionTypes = [...commonTypes, ...ethSpecificTypes];
+
+export const icpTransactionTypes = [...commonTypes, ...icpSpecificTypes];
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/schema/transaction.schema.ts:15]: wondering if all the above shall go in folder `constants`...

**Comment by AntonioVentilii** [comment]
@peterpeterparker please re-review the PR, I changed a bit the definition of the schema, not sure if overkilling

File: src/frontend/src/lib/schema/transaction.schema.ts (hunk 4 above)
Comment (peterpeterparker) [reply src/frontend/src/lib/schema/transaction.schema.ts:15]: what about `z.enum` instead of const?

Comment (AntonioVentilii) [reply src/frontend/src/lib/schema/transaction.schema.ts]: done!

Comment (peterpeterparker) [reply src/frontend/src/lib/schema/transaction.schema.ts]: git push

Comment (peterpeterparker) [reply src/frontend/src/lib/schema/transaction.schema.ts]: (looks like you did not)

Comment (AntonioVentilii) [reply src/frontend/src/lib/schema/transaction.schema.ts]: ah sorry, check now please

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

File: src/frontend/src/lib/schema/transaction.schema.ts (hunk 4 above)
Comment (peterpeterparker) [reply src/frontend/src/lib/schema/transaction.schema.ts]: my bad it was already good, sorry. I think it's good to keep the constant there, they are only meant for zod.

**Comment by AntonioVentilii** [comment]
[OISY-130]

**Pull request merged**